frontend/app/components/work-packages/wp-attachments-formattable-field/wp-attachments-formattable.models.ts

```diff
@@ -0,0 +1,184 @@
+        this.isUpload = this._isUpload(dt);
+        this.isDelayedUpload = this.workPackage.isNew;
+        this.isWebLink = ! this.isUpload;
+        this.webLinkUrl = dt.getData("URL");
+    }
+  
+    public isWebImage(): boolean {
+        if(angular.isDefined(this.webLinkUrl)){
+            return (this.config.imageFileTypes.indexOf(this.webLinkUrl.split(".").pop().toLowerCase()) > -1);
+        }
+    };
+  
+    public isAttachmentOfCurrentWp():boolean {
+        if(this.isWebLink){
+
+            // weblink does not point to our server, so it can't be an attachment
+            if(!(this.webLinkUrl.indexOf(this.$location.host()) > -1) ) return false;
+
+            var isAttachment:boolean = false;
+
+            this.workPackage.attachments.elements.forEach((attachment)=>{
+                if(this.webLinkUrl.indexOf(attachment.href) > -1) {
+                    isAttachment = true;
+                    return;
+                }
+            });
+            return isAttachment;
+        }
+    };
+
+    public filesAreValidForUploading(): boolean {
+      // needs: clarifying if rejected filetypes are a wanted feature
+      // no filetypes are getting rejected yet
+      var allFilesAreValid = true;
+      /*this.files.forEach((file)=>{
+       if(file.size > this.config.maximumAttachmentFileSize) {
+       allFilesAreValid = false;
+       return;
+       }
+       });*/
+      return allFilesAreValid;
+    };
+  
+    public removeHostInformationFromUrl(): string {
+        return this.webLinkUrl.replace(window.location.origin, "");
+    };
+
+    protected _isUpload(dt: DataTransfer): boolean {
+        if (dt.types && this.filesCount > 0) {
+            for (let i=0; i < dt.types.length; i++) {
+                if (dt.types[i] == "Files") {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+}
+
+export class SingleAttachmentModel {
+    protected imageFileExtensions: Array<string> = ['jpeg','jpg','gif','bmp','png'];
+
+    public fileExtension: string;
+    public fileName: string;
+    public isAnImage: boolean;
+    public url: string;
+
+
+    constructor(protected attachment: any){
+        if(angular.isDefined(attachment)){
+            this.fileName = attachment.fileName || attachment.name;
+            this.fileExtension = this.fileName.split(".").pop().toLowerCase();
+            this.isAnImage = this.imageFileExtensions.indexOf(this.fileExtension) > -1;
+            this.url = angular.isDefined(attachment._links) ? attachment._links.downloadLocation.href : '';
+        }
+    }
+
+}
+
+export class FieldModel implements IApplyAttachmentMarkup {
+    public contentToInsert: string;
+
+    constructor(protected workPackage: WorkPackageResource, protected markupModel: MarkupModel){
+        this.contentToInsert = workPackage.description.raw || "";
+    }
+
+
+    private addInitialLineBreak(): string {
+        return (this.contentToInsert.length > 0) ? "\r\n" : "";
```

Comment:
You can use a single `\n` to force a new line when outputting a string. I believe `IE 9` and some version of Opera was the last browser version that required `\r\n` on Windows.